src/components/navs/AppNav/AppNavSettings.vue

```diff
@@ -1,3 +1,78 @@
+
+<script setup lang="ts">
+import { Network } from '@balancer-labs/sdk';
+
+import AppSlippageForm from '@/components/forms/AppSlippageForm.vue';
+import Avatar from '@/components/images/Avatar.vue';
+import useEthereumTxType from '@/composables/useEthereumTxType';
+import { ethereumTxTypeOptions } from '@/constants/options';
+import useWeb3 from '@/services/web3/useWeb3';
+import {
+  getConnectorLogo,
+  getConnectorName,
+} from '@/services/web3/web3.plugin';
+import { shorten } from '@/lib/utils';
+import useDarkMode from '@/composables/useDarkMode';
+
+// COMPOSABLES
+const { darkMode, setDarkMode } = useDarkMode();
+
+const {
+  account,
+  profile,
+  disconnectWallet,
+  toggleWalletSelectModal,
+  connector,
+  provider,
+  isEIP1559SupportedNetwork,
+  userNetworkConfig,
+  isUnsupportedNetwork,
+  explorerLinks,
+} = useWeb3();
+const { ethereumTxType, setEthereumTxType } = useEthereumTxType();
+
+// DATA
+const data = reactive({
+  copiedAddress: false,
+});
+
+// COMPUTED
+const avatarUri = computed(() => profile.value?.avatar || undefined);
```

Comment:
seems no need in `|| undefined`

Reply:
It is needed to avoid `null` value, which is not a valid value for iconUri prop in Avatar component.

I could change the type of that prop but then I should also change this api:

`const iconSRC = computed(() => resolve(props.iconURI));
`
And i prefered to avoid that for now.

Reply:
ah, I see, ok